inference/core/workflows/core_steps/classical_cv/distance_measurement/v1.py

```diff
@@ -0,0 +1,345 @@
+        examples=[2.5, "$inputs.reference_height"],
+        gt=0,
+        json_schema_extra={
+            "relevant_for": {
+                "calibration_type": {
+                    "values": ["reference object"],
+                    "required": True,
+                },
+            },
+        },
+    )
+    
+    pixel_ratio: Union[float, WorkflowParameterSelector(kind=[FLOAT_KIND])]  = Field(
+        title="Reference Pixel-to-Centimeter Ratio",
+        description="The pixel-to-centimeter ratio of the input image, i.e. 1 centimeter = 100 pixels.",
+        default=100,
+        examples=[100, "$inputs.pixel_ratio"],
+        gt=0,
+        json_schema_extra={
+            "relevant_for": {
+                "calibration_type": {
+                    "values": ["pixel-ratio"],
+                    "required": True,
+                },
+            },
+        },
+    )
+
+    @classmethod
+    def describe_outputs(cls) -> List[OutputDefinition]:
+        return [
+            OutputDefinition(name=OUTPUT_KEY, kind=[DICTIONARY_KIND]),
+        ]
+
+    @classmethod
+    def get_execution_engine_compatibility(cls) -> Optional[str]:
+        return ">=1.0.0,<2.0.0"
+
+
+class DistanceMeasurementBlockV1(WorkflowBlock):
+
+    @classmethod
+    def get_manifest(cls) -> Type[WorkflowBlockManifest]:
+        return BlockManifest
+
+    def run(
+        self,
+        predictions: Batch[sv.Detections],
+        object_1_class_name: str,
+        object_2_class_name: str,
+        reference_axis: Literal["horizontal", "vertical"],
+        calibration_type: Literal["reference object", "pixel-ratio"],
+        reference_object_class_name: str,
+        reference_width: float,
+        reference_height: float,
+        pixel_ratio: float,
+    ) -> BlockResult:
+        if calibration_type == "reference object":
+            reference_predictions = predictions
+            distances = measure_distance_with_reference_object(
+                detections=predictions,
+                object_1_class_name=object_1_class_name,
+                object_2_class_name=object_2_class_name,
+                reference_predictions=reference_predictions,
+                reference_object_class_name=reference_object_class_name,
+                reference_width=reference_width,
+                reference_height=reference_height,
+                reference_axis=reference_axis,
+            )
+        elif calibration_type == "pixel-ratio":
+            distances = measure_distance_with_pixel_ratio(
+                detections=predictions,
+                pixel_ratio=pixel_ratio,
+                object_1_class_name=object_1_class_name,
+                object_2_class_name=object_2_class_name,
+                reference_axis=reference_axis,
+            )
+        else:
+            raise ValueError(f"Invalid calibration type: {calibration_type}")
+        
+        return {OUTPUT_KEY: distances}
```

Comment:
I suggest not packing the outputs in inner dicts and manifest multiple separate outputs with primitive values - otherwise to process the output by other blocks (for instance making if-statements) we would need dedicated UQL operations raising the adoption bar

Reply:
Done